### DISCUSSION TITLE
deleteUploadedFileUsing in Builder and handle deletion of files before/after saving the data

### DISCUSSION THREAD

**Discussion by invaders-xx** [filamentphp/filament#2129] [Help]
I have a Builder with a multiple image field. How can I manage the deletion of the images from the disk since the structure and data are saved when the Save button is clicked.

**Comment by ryangjchandler** [comment]
@invaders-xx You'll have to use an observer/model event to look at the changes and handle the deletion yourself.

**Comment by invaders-xx** [comment]
@ryangjchandler, you mean that I will have to go through the json before save and check the difference of data between new and old and delete files on disk or not ?

**Reply by invaders-xx** [reply]
Or I should trigger a save each time an image is deleted ?
